### PR TITLE
feat(product): return parents tree

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
*What*

include option `include_ancestors_tree` to list products including the parent tree.

### CODE REVIEW

File: packages/product/src/repositories/product-category.ts [hunk 1]
```diff
@@ -56,65 +59,118 @@ export class ProductCategoryRepository extends DALUtils.MikroOrmBaseTreeReposito
...
-      )
+    mpaths.push({ mpath: Array.from(parentMpaths) })
 
-      const descendantsByParentId = groupBy(
-        descendantsForCategory,
-        (pc) => pc.parent_category_id
-      )
+    const whereOptions = {
+      ...findOptions.where,
+      $or: mpaths,
```
Comment (riqwan) [inline packages/product/src/repositories/product-category.ts:116]: we'll be overriding existing or conditions here, might be good to merge mpaths with existing.

File: packages/product/src/repositories/product-category.ts [hunk 2]
```diff
@@ -56,65 +59,118 @@ export class ProductCategoryRepository extends DALUtils.MikroOrmBaseTreeReposito
...
-    if (!includeDescendantsTree) {
+    if (!includeDescendantsTree && !includeParentsTree) {
       return productCategories
     }
 
-    return this.buildProductCategoriesWithDescendants(
+    return this.buildProductCategoriesWithTree(
+      {
+        descendants: includeDescendantsTree,
+        parents: includeParentsTree,
```
Comment (riqwan) [inline packages/product/src/repositories/product-category.ts:69]: nit: feel free to ignore - calling this ancestors, to align with descendants.

File: packages/product/integration-tests/__tests__/services/product-category/index.ts [hunk 3]
```diff
@@ -173,6 +170,365 @@ moduleIntegrationTestRunner({
...
+                  parent_category_id: null,
+                  parent_category: null,
+                },
+              },
+              category_children: [
+                {
+                  id: "category-1-b-1",
+                  handle: "category-1-b-1",
+                  mpath: "category-0.category-1.category-1-b.category-1-b-1.",
+                  parent_category_id: "category-1-b",
```
Comment (riqwan) [inline packages/product/integration-tests/__tests__/services/product-category/index.ts:525]: shouldn't there be an empty category_children here?
```suggestion
                  parent_category_id: "category-1-b",
                  category_children: [],
```

**Review by riqwan — APPROVED** [review]
nice! 🥇

File: packages/product/integration-tests/__tests__/services/product-category/index.ts (hunk 3 above)
Comment (carlos-r-l-rodrigues) [reply packages/product/integration-tests/__tests__/services/product-category/index.ts:525]: When both flags are `true` we decided to remove the parent and children properties from categories that are not in the root level. This is tricky to support without dealing with circular references.

**Comment by carlos-r-l-rodrigues** [comment]
@olivermrbl @kasperkristensen @adrien2p can we merge this?

**Review by olivermrbl — COMMENTED** [review]
LGTM

not important but there's a discrepancy in naming ancestors vs. parents

also, can we select fields on ancestors? e.g. if we only want `id` and `name`

**Comment by carlos-r-l-rodrigues** [comment]
It was named parents, beu Riqwan suggested to keep it similar to descendants. I believe everywhere it was named like ancestors, which makes more sense as all of them are returned.

At the moment the fields returned are the same requested to the categories.

**Pull request merged**